druid/src/widget/added.rs

```diff
@@ -0,0 +1,61 @@
+// Copyright 2020 The Druid Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! A [`Controller`] widget that responds to [`LifeCycle::WidgetAdded`] event.
+//!
+//! [`Controller`]: crate::widget::Controller
+//! [`LifeCycle::WidgetAdded`]: crate::LifeCycle::WidgetAdded
+
+use crate::widget::Controller;
+use crate::{Data, Env, LifeCycleCtx, Widget};
+/// This [`Controller`] widget responds to [`LifeCycle::WidgetAdded`] event
+/// with the provided closure. Pass this and a child widget to [`ControllerHost`]
+/// to respond to the event when the child widget is added to the widget tree.
+/// This is also available, for convenience, as an `on_added` method
+/// via [`WidgetExt`].
+///
+/// [`Controller`]: crate::widget::Controller
+/// [`ControllerHost`]: crate::widget::ControllerHost
+/// [`WidgetExt`]: crate::widget::WidgetExt
+/// [`LifeCycle::WidgetAdded`]: crate::LifeCycle::WidgetAdded
+pub struct Added<T> {
+    /// A closure that will be invoked when the child widget is added
+    /// to the widget tree
+    action: Box<dyn Fn(&mut LifeCycleCtx, &T, &Env)>,
```

Comment:
One thing I wonder: should this closure also be given access to the actual *widget*? Can we imagine, for instance, wanting to call some method on the widget from this closure?

Reply:
At the very least it brings it more in line with the `lifecycle` function in `Controller`. It would eliminate the need for the user to still implement `Controller` if they wanted to do something with the child when its added to the widget tree.

Reply:
yea, that's my thinking, I think we should just have the same signature as the method in `Controller`.